src/push-notifications.js

```diff
@@ -175,6 +182,22 @@ class PushNotificationsInstance {
     return this;
   }
 
+  async getState() {
+    if (Notification.permission === 'denied') {
+      return STATE.BLOCKED;
+    }
+
+    if (Notification.permission === 'granted' && this.deviceId !== null) {
+      return STATE.READY_FOR_NOTIFICATIONS;
+    }
+
+    if (Notification.permission === 'granted' && this.deviceId === null) {
+      return STATE.NOT_STARTED_HAS_PERMISSION;
+    }
+
+    return STATE.NOT_STARTED_NEEDS_PERMISSION;
```

Comment:
Not a blocker, but I think I preferred `NOT_STARTED_WILL_PROMPT` or similar because `NEEDS_PERMISSION` is true of `BLOCKED` as well.

Reply:
I agree